libs/core/src/lib/form/form-input-message-group/form-input-message-group.component.ts

```diff
@@ -41,11 +53,24 @@ export class FormInputMessageGroupComponent {
     @Output()
     isOpenChange: EventEmitter<boolean> = new EventEmitter<boolean>();
 
+    /** @hidden */
+    public placement$: Observable<Placement>;
+
+    constructor(@Optional() private _rtlService: RtlService) {
+        this._createRtlObservable();
+    }
+
     /**
      * Function is called every time message changes isOpen attribute
      */
     public openChanged(isOpen: boolean) {
         this.isOpenChange.emit(isOpen);
     }
 
+    /** @hidden */
+    private _createRtlObservable(): void {
+        this.placement$ = this._rtlService
+            ? this._rtlService.rtl.pipe(map(isRtl => isRtl ? 'bottom-end' : 'bottom-start'))
```

Comment:
I think this way is more concise: https://github.com/SAP/fundamental-ngx/blob/master/libs/core/src/lib/breadcrumb/breadcrumb.component.ts#L149

Reply:
I don't agree.
- This example needs to take care of two subscriptions instead of one. (by the way there is memory leak, because RTL service is never unsubscribed).
- This is not a valid usage of Subject [From RXJS creators](https://medium.com/@benlesh/on-the-subject-of-subjects-in-rxjs-2b08b7198b93)

If you want to I can split this example to If/else to be more readable.

Reply:
lets clear things :)
- Subjects are both observer and observable ![source](https://medium.com/@benlesh/on-the-subject-of-subjects-in-rxjs-2b08b7198b93)
- rxjs didn't follow the exact specification of ![observer pattern](https://en.wikipedia.org/wiki/Observer_pattern)
- if subscribe is done in ts file, the unsubscribe method must be called
- if the async pipe is used and there is no subscribe in ts file then async pipe takes care and unsubscribe from the subscription
@mikerodonnell89 I prefer @salarenko approach because there is no need for the state to be stored in the component in fact only source of truth is rtl service. I have changed my implementation.